src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt

```diff
@@ -1,111 +1,88 @@
 package com.hibob.academy.dao
 
-import PetDao
 import com.hibob.academy.utils.BobDbTest
-
 import org.jooq.DSLContext
 import org.junit.jupiter.api.AfterEach
 import org.junit.jupiter.api.Assertions.*
 import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.springframework.beans.factory.annotation.Autowired
 import java.sql.Date
-import java.util.*
-import kotlin.random.Random
-
 
 @BobDbTest
 class PetDaoTest @Autowired constructor(private val sql: DSLContext){
 
     private val petDao = PetDao(sql)
-    private val table = PetTable.instance
+    private val petTable = PetTable.instance
     val companyId = 12L
     val ownerId = null
 
     @Test
     fun `create a new pet that doesn't exist in the database`() {
-        val petName = "Angie"
-        val petTypeString = "Dog"
-        val petType = PetType.convertStringToPetType(petTypeString)
-        val petCompanyId = companyId
-        val dateOfArrival = Date.valueOf("2010-05-20")
-
-        val petId = petDao.createPet(petName, petTypeString, petCompanyId, dateOfArrival, ownerId)
-        val expected = listOf(PetData(petId, petName, petType, petCompanyId, dateOfArrival, ownerId))
+        val pet = PetCreationRequest("Angie", PetType.convertStringToPetType("Dog"), companyId, Date.valueOf("2010-05-20"), ownerId = null )
+        val petId = petDao.createPet(pet)
+        val expected = listOf(PetData(petId, pet.name, pet.type, pet.companyId, pet.dateOfArrival, pet.ownerId))
         assertEquals(expected, petDao.getAllPets(companyId))
     }
 
     @Test
     fun `get pets by type`() {
-        val petName = "Angie"
-        val petTypeString = "Dog"
-        val petCompanyId = companyId
-        val dateOfArrival = Date.valueOf("2010-05-20")
-
-        petDao.createPet(petName, petTypeString, petCompanyId, dateOfArrival, ownerId)
-
+        val pet = PetCreationRequest("Angie", PetType.convertStringToPetType("Dog"), companyId, Date.valueOf("2010-05-20"), ownerId)
```

Comment:
Please add another test that adds both cat and dog and get back the expected by type
Here the test is correct but does not cover the case there are results